src/artworks/artworks.repository.ts

```diff
@@ -62,20 +62,20 @@ export class ArtworksRepository {
     ): Promise<Artwork[]> {
         const orderByPath = ARTWORKS_ORDER_KEY_PATH_MAP[orderBy];
         const results = await buildQuery(this.artworks, limit, orderByPath, direction, filters).get();
-        return Promise.all(results.docs.map(doc => this.toArtwork(withId<FirebaseArtwork>(doc.data(), doc.id))));
+        // @ts-ignore
+        return Promise.all(results.docs.map(async doc => await this.toArtwork({ ...doc.data(), id: doc.id })));
     }
 
     async getArtwork(id: string): Promise<Artwork> {
         const ref = this.artworks.doc(id);
         const doc = await ref.get();
-        return this.toArtwork(withId<FirebaseArtwork>(doc.data(), id));
+        // @ts-ignore
+        return this.toArtwork({ ...doc.data(), id });
```

Comment:
Oh cool, thanks for this! I did not think TS had this feature too. The spread operator is much cleaner, but it's just because TS cannot correctly infer the properties of `doc.data()` - which is why it gives a type error.

I suggest we just assert the type using `as` as a hint to TS that it is a type of `FirebaseArtwork`.

```ts
{ ...doc.data() as FirebaseArtwork, id }
```